db/schema_tables.cc

```diff
@@ -1075,6 +1106,10 @@ static future<> do_merge_schema(distributed<service::storage_proxy>& proxy, std:
     auto&& old_views = co_await read_tables_for_keyspaces(proxy, keyspaces, views());
     auto old_functions = co_await read_schema_for_keyspaces(proxy, FUNCTIONS, keyspaces);
     auto old_aggregates = co_await read_schema_for_keyspaces(proxy, AGGREGATES, keyspaces);
+    auto old_scylla_aggregates = schema_result();
+    if (proxy.local().features().cluster_schema_features().contains<schema_feature::SCYLLA_KEYSPACES>()) {
+        old_scylla_aggregates = co_await read_schema_for_keyspaces(proxy, SCYLLA_AGGREGATES, keyspaces);
+    }
```

Comment:
This can (should?) be unconditional. schema_features are about what we can export to other nodes. Locally we're always self-consistent - if the code has support for the new table, then it exists.

I can't think of a concrete reason to make it unconditional (if the cluster hasn't agreed on the schema feature, then the table should be empty), it just seems less fragile to me, to avoid races during boot.

btw, why are you checking schema_featrure::SCYLLA_KEYSPACES for the SCYLLA_AGGREGATES table?

Reply:
fixed